### PR TITLE
Update libraries, dependencies, run eslint

### PR DESCRIPTION
**Pull request by mirpo**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "streamr-client",
-  "version": "2.1.1",
+  "version": "2.1.0",
```
Comment (harbu) [inline package.json]: ?

File: src/Connection.js [hunk 2]
```diff
@@ -25,7 +24,7 @@ class Connection extends EventEmitter {
     connect() {
         if (this.state === Connection.State.CONNECTING) {
             return Promise.reject(new Error('Already connecting!'))
-        } else if (this.state === Connection.State.CONNECTED) {
+        } if (this.state === Connection.State.CONNECTED) {
```
Comment (harbu) [inline src/Connection.js]: newline in between

File: src/Connection.js [hunk 3]
```diff
@@ -86,9 +85,9 @@ class Connection extends EventEmitter {
     disconnect() {
         if (this.state === Connection.State.DISCONNECTING) {
             return Promise.reject(new Error('Already disconnecting!'))
-        } else if (this.state === Connection.State.DISCONNECTED) {
+        } if (this.state === Connection.State.DISCONNECTED) {
```
Comment (harbu) [inline src/Connection.js]: newline in between

File: src/Connection.js [hunk 4]
```diff
@@ -86,9 +85,9 @@ class Connection extends EventEmitter {
...
             return Promise.reject(new Error('Already disconnected!'))
-        } else if (this.socket === undefined) {
+        } if (this.socket === undefined) {
```
Comment (harbu) [inline src/Connection.js]: newline in between

File: src/Signer.js [hunk 5]
```diff
@@ -52,9 +53,9 @@ export default class Signer {
             if (msg.prevMsgRef) {
                 prev = `${msg.prevMsgRef.timestamp}${msg.prevMsgRef.sequenceNumber}`
             }
-            return `${msg.getStreamId()}${msg.getStreamPartition()}${msg.getTimestamp()}${msg.messageId.sequenceNumber}` +
-                `${address.toLowerCase()}${msg.messageId.msgChainId}${prev}${msg.getSerializedContent()}`
-        } else if (signatureType === SIGNATURE_TYPES.ETH_LEGACY) {
+            return `${msg.getStreamId()}${msg.getStreamPartition()}${msg.getTimestamp()}${msg.messageId.sequenceNumber}`
+                + `${address.toLowerCase()}${msg.messageId.msgChainId}${prev}${msg.getSerializedContent()}`
+        } if (signatureType === SIGNATURE_TYPES.ETH_LEGACY) {
```
Comment (harbu) [inline src/Signer.js]: newline in between

File: src/Signer.js [hunk 6]
```diff
@@ -79,9 +80,9 @@ export default class Signer {
     static createSigner(options, publishWithSignature) {
         if (publishWithSignature === 'never') {
             return undefined
-        } else if (publishWithSignature === 'auto' && !options.privateKey && !options.provider) {
+        } if (publishWithSignature === 'auto' && !options.privateKey && !options.provider) {
```
Comment (harbu) [inline src/Signer.js]: newline in between

File: src/Signer.js [hunk 7]
```diff
@@ -79,9 +80,9 @@ export default class Signer {
...
             return undefined
-        } else if (publishWithSignature === 'auto' || publishWithSignature === 'always') {
+        } if (publishWithSignature === 'auto' || publishWithSignature === 'always') {
```
Comment (harbu) [inline src/Signer.js]: newline in between

File: src/StreamrClient.js [hunk 8]
```diff
@@ -293,7 +293,7 @@ export default class StreamrClient extends EventEmitter {
         if (this.isConnected()) {
             // If connected, emit a publish request
             return this._requestPublish(streamMessage, sessionToken)
-        } else if (this.options.autoConnect) {
+        } if (this.options.autoConnect) {
```
Comment (harbu) [inline src/StreamrClient.js]: newline in between

File: src/StreamrClient.js [hunk 9]
```diff
@@ -470,7 +470,7 @@ export default class StreamrClient extends EventEmitter {
     connect() {
         if (this.isConnected()) {
             return Promise.reject(new Error('Already connected!'))
-        } else if (this.connection.state === Connection.State.CONNECTING) {
+        } if (this.connection.state === Connection.State.CONNECTING) {
```
Comment (harbu) [inline src/StreamrClient.js]: newline in between

File: src/SubscribedStream.js [hunk 10]
```diff
@@ -23,7 +23,7 @@ export default class SubscribedStream {
                 return Signer.verifyStreamMessage(msg, new Set(publishers))
             }
             return false
-        } else if (this._client.options.verifySignatures === 'never') {
+        } if (this._client.options.verifySignatures === 'never') {
```
Comment (harbu) [inline src/SubscribedStream.js]: newline in between

File: src/rest/StreamEndpoints.js [hunk 11]
```diff
@@ -22,7 +22,7 @@ const agentByProtocol = {
 function getKeepAliveAgentForUrl(url) {
     if (url.startsWith('https')) {
         return agentByProtocol.https
-    } else if (url.startsWith('http')) {
+    } if (url.startsWith('http')) {
```
Comment (harbu) [inline src/rest/StreamEndpoints.js]: newline in between

File: package.json (hunk 1 above)
Comment (mirpo) [reply package.json]: 👍

File: src/rest/StreamEndpoints.js (hunk 11 above)
Comment (mirpo) [reply src/rest/StreamEndpoints.js]: Added rule for newline between if

**Pull request merged**